### PR TITLE
[Feature] Add transport client for SSE-based MCP servers

### PR DESCRIPTION
**Pull request by noahlozevski**
This pull request finalizes the **SSEClientTransport** class, enabling the MCP client to connect via **(SSE)**. Key improvements include:

1. **Downchannel Connection**  
   - On client startup, an asynchronous SSE GET request (the “downchannel”) is established.  
   - An `event: endpoint` message from the server informs the client of the correct POST URL.  
   - The client sets `postEndpoint` only when the server sends this `endpoint` event.  

2. **Data Serialization & POST**  
   - The client can now send data back to the SSE server over HTTP POST once `postEndpoint` is known.  
   - If no endpoint is received, an error is raised when attempting to send.  

3. **Error Handling & Graceful Termination**  
   - If the SSE stream fails (network error, unexpected status code, etc.), the client closes its message stream with an error.  
   - If the server or client terminates normally, the transport finishes gracefully without throwing.  

4. **Test Coverage**  
   - Introduces a **Node.js-based SSE server** (`sse.js`) for local integration tests.  
   - The Swift test suite spawns this server (requires `node` in the PATH) and exercises typical scenarios:
     - Basic connection and receiving the `endpoint` event  
     - Sending messages (POST) to the server  
     - Handling errors and server-initiated disconnects  
     - Changing endpoints mid-connection  
   - Verifies that the client transitions to `.connected`, updates `postEndpoint` as expected, and handles edge cases.  

Other notes
- I removed the `suspended` transport state. i couldnt think of when this would be needed and i thought it was useless
- Need to hook this up and test this with a real server

### CODE REVIEW

File: Sources/SwiftMCP/Transport/SSETransport.swift [hunk 1]
```diff
@@ -56,72 +68,154 @@ public actor SSETransport: MCPTransport {
...
+
+                // SSE lines typically come in the form of:
+                // event: <something>
+                // data: <something>
+                // or
+                // data: <json>
+                // No whitespace rules are technically enforced, but we trim for whitespace on the endpoint event
+
+                for try await line in asyncBytes.lines {
+                    print("NOAH : \(line)")
```
Comment (gavinaboulhosn) [inline Sources/SwiftMCP/Transport/SSETransport.swift:166]: nit - remove debug prints

**Review by gavinaboulhosn — APPROVED** [review]
